common/src/main/java/com/wynntils/features/user/overlays/MiniMapOverlayFeature.java

```diff
@@ -0,0 +1,189 @@
+    public final MiniMapOverlay miniMapOverlay = new MiniMapOverlay();
+
+    public static class MiniMapOverlay extends Overlay {
+
+        @Config
+        public float scale = 1f;
+
+        @Config
+        public boolean followPlayerRotation = true;
+
+        @Config
+        public boolean renderUsingLinear = true;
+
+        @Config
+        public MapMaskType maskType = MapMaskType.Rectangular;
+
+        public MiniMapOverlay() {
+            super(
+                    new OverlayPosition(
+                            0,
+                            0,
+                            VerticalAlignment.Top,
+                            HorizontalAlignment.Left,
+                            OverlayPosition.AnchorSection.TopLeft),
+                    new GuiScaledOverlaySize(200, 200));
+        }
+
+        @Override
+        public void render(PoseStack poseStack, float partialTicks, Window window) {
+            if (!WebManager.isMapLoaded()) return;
+
+            MapProfile map = WebManager.getMaps().get(0);
+
+            float uScale = 1f / map.getTextureWidth();
+            float vScale = 1f / map.getTextureHeight();
+
+            float width = getWidth();
+            float height = getHeight();
+            float renderX = getRenderX();
+            float renderY = getRenderY();
+
+            float centerX = renderX + width / 2;
+            float centerZ = renderY + height / 2;
+            float textureX = map.getTextureXPosition(McUtils.player().getX());
+            float textureZ = map.getTextureZPosition(McUtils.player().getZ());
+
+            // avoid rotational overpass
+            float extraFactor = (followPlayerRotation && maskType == MapMaskType.Rectangular ? 1.5F : 1);
+
+            float halfRenderedWidth = width / 2 * extraFactor;
+            float halfRenderedHeight = height / 2 * extraFactor;
+            float halfTextureWidth = halfRenderedWidth * scale;
+            float halfTextureHeight = halfRenderedHeight * scale;
+
+            // Render Minimap
+            RenderSystem.setShader(GameRenderer::getPositionTexShader);
+            RenderSystem.setShaderTexture(0, map.resource());
+
+            // clamp map rendering
+            int option = renderUsingLinear ? GL11.GL_LINEAR : GL11.GL_NEAREST;
+            RenderSystem.texParameter(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_MAG_FILTER, option);
+
+            RenderSystem.texParameter(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_T, GL13.GL_CLAMP_TO_BORDER);
+            RenderSystem.texParameter(GL11.GL_TEXTURE_2D, GL11.GL_TEXTURE_WRAP_S, GL13.GL_CLAMP_TO_BORDER);
+
+            // enable mask
+            switch (maskType) {
+                case Rectangular -> {
+                    RenderUtils.enableScissor((int) renderX, (int) renderY, (int) getWidth(), (int) getHeight());
+                }
+                case Circle -> {
+                    // TODO
+                }
+            }
+            {
```

Comment:
These braced blocks with no condition/loop specifier looks a bit odd. I understand that you wanted to group the code, but I suggest you do it either by extracting functionality into a separate private method, or by using empty line and judiciously placed line commends (they can even include markers like `// -----`).

Reply:
Will do